sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala

```diff
@@ -100,6 +101,37 @@ private[spark] object SQLConf {
         }
       }, _.toString, doc, isPublic)
 
+    def intMemConf(
+        key: String,
+        defaultValue: Option[Int] = None,
+        doc: String = "",
+        isPublic: Boolean = true): SQLConfEntry[Int] =
+      SQLConfEntry(key, defaultValue, { v =>
+        val isInteger: Option[Int] = try {
```

Comment:
I find this body simpler:

```
try {
  v.toInt
} catch {
  case _: NumberFormatException =>
    val sizeInBytes = try {
      Utils.byteStringAsBytes(v)
    } catch {
      case _: NumberFormatException =>
        throw new IllegalArgumentException(s"$key should be int, but was $v")
    }
    require(getSizeAsBytes <= Int.MaxValue && getSizeAsBytes >= Int.MinValue)
    sizeInBytes.toInt
}
```

I'd like to hear from @marmbrus or @viirya for example before merging a change like this.